tensortrade/exchanges/simulated/simulated_exchange.py

```diff
@@ -146,67 +162,106 @@ def transform_data_frame(self) -> bool:
-            self._trades = self._trades.append({
-                'step': self._current_step,
-                'symbol': trade.symbol,
-                'type': trade.trade_type,
-                'amount': trade.amount,
-                'price': trade.price
-            }, ignore_index=True)
+        return False
 
+    def _update_account(self, trade: Trade):
+        log = {'step': self._current_step}
         if trade.is_buy:
-            self._balance -= trade.amount * trade.price
-            self._portfolio[trade.symbol] = self._portfolio.get(trade.symbol, 0) + trade.amount
+            log['action'] = trade.log
+            self._balance -= trade.transact_total
+            self._portfolio[trade.symbol] = self._portfolio.get(trade.symbol, 0) + trade.transact_amount
         elif trade.is_sell:
-            self._balance += trade.amount * trade.price
-            self._portfolio[trade.symbol] = self._portfolio.get(trade.symbol, 0) - trade.amount
+            log['action'] = trade.log
+            self._balance += trade.transact_total
+            self._portfolio[trade.symbol] = self._portfolio.get(trade.symbol, 0) - trade.transact_amount
+        elif trade.is_hold:
+            log['action'] = trade.log
+        else:
+            log['action'] = "Unknown Trade Type: {}".format(trade.to_dict)
+
+        if self._is_valid_trade(trade) and not trade.is_hold:
+            self._trades = self._trades.append({'step': self._current_step,
+                                                'symbol': trade.symbol,
+                                                'type': trade.trade_type,
+                                                'amount': trade.transact_amount,
+                                                'price': trade.transact_total}, ignore_index=True)
 
         self._portfolio[self._base_instrument] = self._balance
 
-        self._performance = self._performance.append({
-            'balance': self.balance,
-            'net_worth': self.net_worth,
-        }, ignore_index=True)
+        log.update({'balance': self.balance,
+                    'net_worth': self.net_worth
+                    })
+        self._performance = self._performance.append( log , ignore_index=True)
 
     def execute_trade(self, trade: Trade) -> Trade:
-        current_price = self.current_price(symbol=trade.symbol)
-        commission = self._commission_percent / 100
-        filled_trade = trade.copy()
-
-        if filled_trade.is_hold or not self._is_valid_trade(filled_trade):
-            filled_trade.amount = 0
+        trade = trade.copy()
+        transact_amount = trade.order_amount
+        transact_price = trade.order_price
 
-            return filled_trade
+        if trade.is_hold:
+            # hodl boyz!
+            trade.transact_amount = transact_amount
+            trade.transact_price = transact_price
+        else:
+            trade.order_commission = transact_price * self.commission_percent
```

Comment:
Why is `trade.order_commission` calculated here? It is incorrectly calculated (missing `transact_amount * transact_price`), and the correct amount/price has not been set yet since we know the trade is not a `HOLD`.

I believe this entire `else:` statement should be removed.